lib/datadog/core/telemetry/metric.rb

```diff
@@ -41,6 +41,18 @@ def to_h
             }
           end
 
```

Comment:
In implementing equality for all the `Datadog::Core::Telemetry::Event` in `event.rb`, the `Datadog::Core::Telemetry::Metric` objects also needed their equality implemented since some events contain metrics, which need to be compared.